Tests/AWSSDKSwiftCoreTests/AWSRequestTests.swift

```diff
@@ -0,0 +1,320 @@
+        struct Input: AWSEncodableShape {
+            public static let _xmlNamespace: String? = "https://test.amazonaws.com/doc/2020-03-11/"
+            let number: Int
+        }
+        let input = Input(number: 5)
+        let xmlConfig = createServiceConfig(serviceProtocol: .restxml)
+        var request: AWSRequest?
+        XCTAssertNoThrow(request = try AWSRequest(operation: "Test", path: "/", httpMethod: "GET", input: input, configuration: xmlConfig))
+        guard case .xml(let element) = request?.body else {
+            return XCTFail("Shouldn't get here")
+        }
+        XCTAssertEqual(element.xmlString, "<Input xmlns=\"https://test.amazonaws.com/doc/2020-03-11/\"><number>5</number></Input>")
+    }
+
+
+    func testCreateWithPayloadAndXMLNamespace() {
+        struct Payload: AWSEncodableShape {
+            public static let _xmlNamespace: String? = "https://test.amazonaws.com/doc/2020-03-11/"
+            let number: Int
+        }
+        struct Input: AWSEncodableShape & AWSShapeWithPayload {
+            public static let _payloadPath: String = "payload"
+            let payload: Payload
+        }
+        let input = Input(payload: Payload(number: 5))
+        let xmlConfig = createServiceConfig(serviceProtocol: .restxml)
+        var request: AWSRequest?
+        XCTAssertNoThrow(request = try AWSRequest(operation: "Test", path: "/", httpMethod: "GET", input: input, configuration: xmlConfig))
+        guard case .xml(let element) = request?.body else {
+            return XCTFail("Shouldn't get here")
+        }
+        XCTAssertEqual(element.xmlString, "<Payload xmlns=\"https://test.amazonaws.com/doc/2020-03-11/\"><number>5</number></Payload>")
+    }
+
+    func testDataInJsonPayload() {
+        struct DataContainer: AWSEncodableShape {
+            let data: Data
+        }
+        struct J: AWSEncodableShape & AWSShapeWithPayload {
+            public static let _payloadPath: String = "dataContainer"
+            let dataContainer: DataContainer
+        }
+        let input = J(dataContainer: DataContainer(data: Data("test data".utf8)))
+        let jsonConfig = createServiceConfig(serviceProtocol: .json(version: "1.1"))
+        XCTAssertNoThrow(try AWSRequest(operation: "PutRecord",path: "/",httpMethod: "POST", input: input, configuration: jsonConfig))
+    }
+
+    func testEC2ClientRequest() {
+        struct Input: AWSEncodableShape {
+            let array: [String]
+        }
+        let input = Input(array: ["entry1", "entry2"])
+        let config = createServiceConfig(serviceProtocol: .ec2, apiVersion: "2013-12-02")
+        var request: AWSRequest?
+        XCTAssertNoThrow(request = try AWSRequest(operation: "Test", path: "/", httpMethod: "GET", input: input, configuration: config))
+        XCTAssertEqual(request?.body.asString(), "Action=Test&Array.1=entry1&Array.2=entry2&Version=2013-12-02")
+    }
+    
+    func testMiddlewareIsOnlyAppliedOnce() throws {
```

Comment:
Don't think the `testMiddlewareIsOnlyAppliedOnce` test is necessary anymore. It was really there to ensure the `AWSClient` wasn't applying the middleware twice. I think we'll need to do a more complex version of this using the `AWSTestServer` to verify this and at that point the test should be in AWSClient.

Remove this and add issue to write new version